spec/services/user_subscriptions/create_spec.rb

```diff
@@ -0,0 +1,61 @@
+require "rails_helper"
+
+RSpec.describe UserSubscriptions::Create, type: :service do
+  let(:subscriber) { create(:user) }
+
+  it "returns an error for an invalid source type" do
+    source = create(:comment)
+    user_subscription_params = { source_type: source.class.name, source_id: source.id, subscriber_email: subscriber.email }
+    user_subscription = described_class.call(subscriber, user_subscription_params)
+
+    expect(user_subscription.data).to be_nil
+    expect(user_subscription.success).to be false
+    expect(user_subscription.error).to eq "Invalid source_type."
+  end
+
+  it "returns an error for an invalid source" do
+    source = create(:article, :with_user_subscription_tag_role_user, with_user_subscription_tag: true)
+    user_subscription_params = { source_type: source.class.name, source_id: source.id + 999, subscriber_email: subscriber.email }
```

Comment:
`source_id: source.id + 999`

Is this just a way to create an invalid source to force the validation to fail?

Reply:
Correct!